anheumai-dparys/context/context_test.go

```diff
@@ -0,0 +1,97 @@
+package ctx_is_king
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+type SpyStore struct {
+	response string
+	t        *testing.T
+}
+
+func (s *SpyStore) Fetch(ctx context.Context) (string, error) {
+	data := make(chan string, 1)
+
+	go func() {
+		var result string
+		for _, c := range s.response {
+			select {
+			case <-ctx.Done():
+				s.t.Log("they cancelled why, why???")
```

Comment:
Why? Who? What? The big questions of the universe. Hopefully they will be answered one day. But not today :-)